lib/src/validationEngine.ts

```diff
@@ -79,16 +85,16 @@ export class ValidationEngine implements IValidationEngine {
     return fullFormValidatedPromise;
   }
 
-  triggerFieldValidation(formViewModel: any, key: string, value: any, filter: any = consts.defaultFilter): Promise<FieldValidationResult> {
+  triggerFieldValidation(vm: any, key: string, value: any, filter: any = consts.defaultFilter): Promise<FieldValidationResult> {
     // updated dirty flag and perform validation
     this._isFormPristine = false;
-    return this.validateSingleField(formViewModel, key, value, filter);
+    return this.validateSingleField(vm, key, value, filter);
   }
 
-  validateGlobalFormValidations(vm: any): Array<Promise<FieldValidationResult>> {
+  validateGlobalFormValidations(vm: any) {
```

Comment:
We need to discuss here if we should declare it for functions that already returns that value, e.g:

```ts
function isNotNull(object: any): boolean {
  return object !== null;
}

function giveMeAPromiseWIthFieldValidationResult(someData: any): Promise<ValidationResult> {
  return new Promise((resolve, reject) => {
    const validationResult = new FieldValidationResult();
    // do some logic here
    resolve(validationResult);
  });
}
```